### ISSUE TITLE
Natsue - Unlisted JDBC dependencies?

### ISSUE THREAD

**Issue by TeneDoringo** [20kdc/c3ds-projects#2]
The Maven POM lists no dependencies, however, the Natsue project appears to require either the Sqlite JDBC (by default, and in my case) or the MySQL JDBC:

`class natsue.server.Main: Started logger.
class natsue.server.Main: Read configuration.
natsue.server.database.jdbc.JDBCNatsueDatabase@2e5d6d97: JDBCNatsueDatabase, configured for sqlite
Exception in thread "main" java.sql.SQLException: No suitable driver found for jdbc:sqlite:natsue.db
	at java.sql/java.sql.DriverManager.getConnection(DriverManager.java:702)
	at java.sql/java.sql.DriverManager.getConnection(DriverManager.java:251)
	at natsue.server.database.jdbc.JDBCNatsueDatabase.<init>(JDBCNatsueDatabase.java:41)
	at natsue.server.Main.main(Main.java:55)
`
In addition, the pom.xml doesn't list either as a classpath. I am having difficulties making the server use the Sqlite JDBC jar...

Environment is Linux Mint using OpenJDK Java 11 Runtime. Sqlite version 3.31.1-4ubuntu0.4 is installed. I am attempting to use sqlite-jdbc-3.39.3.0.

**Comment by TeneDoringo** [comment]
When I was tinkering around with trying to get this to work I added it as a dependency to the POM as so:
```
<dependencies>
        <dependency> 
            <groupId>org.xerial</groupId>
            <artifactId>sqlite-jdbc</artifactId>
            <version>3.39.3.0</version> 
        </dependency>
</dependencies>
```
However, while maven does properly resolve the dependency, the server still fails to find the jdbc driver.

**Comment by TeneDoringo** [comment]
Recent breakthrough: I managed to get the server working by changing the plugin used from maven-jar-plugin to maven-assembly-plugin to build the jar with the sqlite JDBC. (Please excuse the messy xml):

```
			<plugin>
				<groupId>org.apache.maven.plugins</groupId>
				<artifactId>maven-assembly-plugin</artifactId>
				 <executions>
                    <execution>
                        <phase>package</phase>
                            <goals>
                                <goal>single</goal>
                            </goals>
                            <configuration>
                                <archive>
                                    <manifest>
                                         <mainClass>
                                            natsue.server.Main
                                         </mainClass>
                                    </manifest>
                                </archive>
                                <descriptorRefs>
                                    <descriptorRef>jar-with-dependencies</descriptorRef>
                                </descriptorRefs>
                             </configuration>
                     </execution>
                   </executions>
            </plugin>
```

This fixes it for my use case, though I do not know if this is a particularly efficient solution. I will be leaving this issue open for your consideration.

**Comment by 20kdc** [comment]
While this issue isn't actually a bug (the idea is that you're supposed to be able to use whatever JDBC driver you want, you just have to hope the SQL works for the target server) -

I *am* going to fix this regardless, though I will be making my own adjustments to the `pom.xml`.

**Comment by 20kdc** [comment]
Resolved as of d9964aeb0a27140560e7f213f04c666558c793af (I hope), reopen if incorrect
